src/main/java/uk/gov/pay/api/service/PaymentSearchService.java

```diff
@@ -0,0 +1,172 @@
+                    .request()
+                    .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                    .get();
+
+            logger.info("response from connector form charge search: " + connectorResponse);
+
+            if (connectorResponse.getStatus() == SC_OK) {
+                try {
+                    JsonNode responseJson = connectorResponse.readEntity(JsonNode.class);
+                    logger.debug("json response from connector from charge search: " + responseJson);
+
+                    TypeReference<PaymentSearchResponse> typeRef = new TypeReference<PaymentSearchResponse>() {};
+                    PaymentSearchResponse searchResponse = objectMapper.readValue(responseJson.traverse(), typeRef);
+                    List<PaymentForSearchResult> chargeFromResponses = searchResponse.getPayments()
+                            .stream()
+                            .map(charge -> PaymentForSearchResult.valueOf(
+                                    charge,
+                                    paymentUriGenerator.getPaymentURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentEventsURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentCancelURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentRefundsURI(baseUrl, charge.getChargeId())))
+                            .collect(Collectors.toList());
+                    HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation.builder()
+                            .addProperty("results", chargeFromResponses)
+                            .addProperty("count", searchResponse.getCount())
+                            .addProperty("total", searchResponse.getTotal())
+                            .addProperty("page", searchResponse.getPage());
+                    addLink(halRepresentation, "self", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getSelf()));
+                    addLink(halRepresentation, "first_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getFirstPage()));
+                    addLink(halRepresentation, "last_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getLastPage()));
+                    addLink(halRepresentation, "prev_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getPrevPage()));
+                    addLink(halRepresentation, "next_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getNextPage()));
+                
+                    return halRepresentation.build();
+                } catch (IOException | ProcessingException | URISyntaxException ex) {
+                    throw new SearchChargesException(ex);
+                }
+            }
+            throw new SearchChargesException(connectorResponse);
+        }
+        throw new NotImplementedException("DD serch is not yet implemented");
```

Comment:
spelling error `search`

Reply:
Also should expand into Direct Debit as few people know that DD refers to Direct Debit

Reply:
It's for one iteration but will change it